providers/mlx5/qp.c

```diff
@@ -942,8 +942,7 @@ static inline int _mlx5_post_send(struct ibv_qp *ibqp, struct ibv_send_wr *wr,
 			mlx5_bf_copy(bf->reg + bf->offset, (unsigned long long *)ctrl,
 				     align(size * 16, 64), qp);
 		else
-			mlx5_write64((__be32 *)ctrl, bf->reg + bf->offset,
-				     &ctx->lock32);
+			mmio_write64_be(bf->reg + bf->offset, *(__be64 *)ctrl);
```

Comment:
No, it is not missed.

It is very obfuscated, but MLX5_PAIR_TO_64  is exactly equivalent to *(uint64 t*)val, eg a native 64 bit load. This means that mlx5_write64() is identical to mmio_write64_be(). 

Eg consider this program on a LE machine:

```
#include <stdint.h>
#include <assert.h>

int main(int argc,const char *argv[])
{
	uint8_t bytes[8] = {0,1,2,3,4,5,6,7};
	uint32_t *bytes32 = bytes;

	assert(*(uint64_t *)bytes == 0x0706050403020100);
	assert(bytes32[1] == 0x07060504);
	assert(bytes32[0] == 0x03020100);
	assert(((uint64_t) bytes32[1] << 32 | bytes32[0]) == 0x0706050403020100);
}
```

And this one on a BE machine:

```
int main(int argc,const char *argv[])
{
	uint8_t bytes[8] = {0,1,2,3,4,5,6,7};
	uint32_t *bytes32 = bytes;

	assert(*(uint64_t *)bytes == 0x0001020304050607);
	assert(bytes32[1] == 0x04050607);
	assert(bytes32[0] == 0x00010203)
	assert(((uint64_t) bytes32[0] << 32 | bytes32[1]) == 0x0001020304050607);
}
```

The native load and the macro produce identical results.